superset/databases/commands/create.py

```diff
@@ -63,8 +67,11 @@ def run(self) -> Model:
             security_manager.add_permission_view_menu("database_access", database.perm)
             db.session.commit()
         except DAOCreateFailedError as ex:
-            logger.exception(ex.exception)
-            raise DatabaseCreateFailedError()
+            with event_logger(
+                action=f"db_creation_failed.{ex.__class__.__name__}",
+                engine=database.db_engine_spec.__name__,
+            ):
+                raise DatabaseCreateFailedError()
```

Comment:
Total nit, but in places where we just raise an exception inside the context manager like here and in lines 94-97 I think it would make more sense to use the `log_with_context` function instead:

```suggestion
            event_logger.log_with_context(
                action=f"db_creation_failed.{ex.__class__.__name__}",
                engine=database.db_engine_spec.__name__,
            )
            raise DatabaseCreateFailedError()
```